### PR TITLE
# ♻️ Refactor Tokens: CORE restructure

### PR DESCRIPTION
**Pull request by 3dyonic**
https://monday.monday.com/boards/245345663/pulses/2427141500/posts/1437071535

* Create Core structure based on hierarchy and scope.
* Create Core Tokens based on scopes.
* Create Core Maps based on scopes, using tokens as values.
* Create Functions: extract rgb, map-collect.
* Create Mixins: generate-custom-properties.
* Generate Core custom properties from core-settings-map.
* Remove Core legacy file duplications.
* Fix themes shadows to include core shadow tokens.

### CODE REVIEW

File: src/core/tokens/_core-shadows.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// CORE TOKENS: SHADOWS
+// This partial defines global shadow related tokens
+
+$shadow-size-small: 0px 4px 8px;
+$shadow-size-medium: 0px 6px 20px;
+$shadow-size-large: 0px 15px 50px;
+
+$shadow-small-opacity: $opacity-20;
+$shadow-medium-opacity: $shadow-small-opacity;
```
Comment (orrgottlieb) [inline src/core/tokens/_core-shadows.scss:9]: why use small ?

File: src/_core.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// GLOBAL SCOPE LOADER: CORE
+// This partial loads Core scope related Partials.
+// - This partials Generates Core custom properties using a mixin and a settings map.
+//   - Mixin is "generate-custom-properties".
+//   - Map is "$core-settings-map" which is a map collected from all other core scope maps.
+// - Core properties are Global and Generic and goes under :root
+
+@import "core/core-tokens";
```
Comment (orrgottlieb) [inline src/_core.scss:8]: why do we need to import other files besides - core-settings

File: src/core/_core-settings.scss [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// CORE MAP : SETTINGS
+// This partial Defines a Unified settings map for the Core Scope.
+// - Map collect function is used in order to merge multiple maps.
+
+$core-settings-map: map-collect(
```
Comment (orrgottlieb) [inline src/core/_core-settings.scss:5]: why not import the relevant maps here?

File: src/core/tokens/_core-typography.scss [hunk 4]
```diff
@@ -0,0 +1,50 @@
+// CORE TOKENS: TYPOGRAPHY
+// This partial Defines Global Typography tokens
+
+// FONTS SCALES
+// - Font Size scale
+$font-size-10: $font-reg-size-14;
```
Comment (orrgottlieb) [inline src/core/tokens/_core-typography.scss:6]: why are we creating these scale variables?

File: src/core/tokens/_core-typography.scss [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+$font-line-height-10: $font-reg-line-height-18;
+$font-line-height-20: $font-reg-line-height-24;
+$font-line-height-30: $font-reg-line-height-24;
+$font-line-height-40: $font-reg-line-height-24;
+$font-line-height-50: $font-reg-line-height-32;
+$font-line-height-60: $font-reg-line-height-52;
+
+// TYPOGRAPHY TOKENS
+// - Font sizes (meaningful units)
+$font-size-h1: var(--font-size-60);
```
Comment (orrgottlieb) [inline src/core/tokens/_core-typography.scss:23]: we want to expose as little as variables as we can - `--font-size-##` will be available for everyone to use and it doesn't imply the developer when to use it

File: src/core/tokens/_core-typography.scss [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+$font-size-h2: var(--font-size-50);
+$font-size-h3: var(--font-size-50);
+$font-size-h4: var(--font-size-40);
+$font-size-h5: var(--font-size-30);
+$font-size-general-label: var(--font-size-20);
+$font-size-paragraph: var(--font-size-30);
+$font-size-subtext: var(--font-size-10);
+
+// - Line heights (meaningful units)
+$font-line-height-h1: var(--font-line-height-60);
```
Comment (orrgottlieb) [inline src/core/tokens/_core-typography.scss:33]: same comment regarding scale variables as custom properites

**Review by orrgottlieb — CHANGES_REQUESTED** [review]

File: src/core/tokens/_core-typography.scss (hunk 5 above)
Comment (3dyonic) [reply src/core/tokens/_core-typography.scss:23]: this is the case right now. 
i did not change it, just wrote it in a token that is generate from a map using a mixin.
the output is the same.

And i agree , this should not be exposed.
Yet in my scope i did not change the output besides removing duplication and changing sort order in some cases.

File: src/core/tokens/_core-typography.scss (hunk 6 above)
Comment (3dyonic) [reply src/core/tokens/_core-typography.scss:33]: same reply ;)

File: src/_core.scss (hunk 2 above)
Comment (3dyonic) [reply src/_core.scss:8]: - We use tokens to registraion values.
- we use map to key value tokens
- we use settings to collect the maps and generate the custom properties from the maps.

- maps have tokens dependencies.
- setting have map dependencies.
- i want to manage all dependencies in a loder (one place).
- This is a convention for global scope CSS.

File: src/core/tokens/_core-typography.scss (hunk 4 above)
Comment (3dyonic) [reply src/core/tokens/_core-typography.scss:6]: this is the case right now.
i did not change it, just wrote it in a token that is generate from a map using a mixin.
the output is the same.

And i agree , this should not be exposed.
Yet in my scope i did not change the output besides removing duplication and changing sort order in some cases.

**Review by orrgottlieb — APPROVED** [review]

**Pull request merged**